### ISSUE TITLE
My logsigmoid seems not convergent with torch.nn.Linear(2048,*)

### ISSUE THREAD

**Issue by lyksdu** [pytorch/pytorch#15279]
## ❓ My logsigmoid seems not convergent with torch.nn.Linear(2048,*)

i am using pytorch 1.0(cpu no cuda), python 3.7

i random three 2048-D Tensors`( i , j , k )`,

first a Linear layer transfer these Tensor to 128-D Tensor (seem others are also exist this problem)

second, calculate `loss =-logsigmoid(i*j  - i*k)'

but, when `i*j - i*k < 0` , the loss can increase rapidly, `i*j - i*k>0`, the loss can decrease slowly

Theoretically,the loss will be lower in any condition , isn't it?

here is a simple demo:

    import torch as T
    from torch.nn import Module, functional as F, Linear
    from torch.optim import SGD, Adam
    vis = 2048
    i= T.randn([vis],requires_grad=True)
    y= Linear(vis,128)
    j= T.randn([vis])
    k= T.randn([vis])
    for i in range(30):
    
        y = -F.logsigmoid((W(i)).unsqueeze(0).mm((W(j)).unsqueeze(0).t())- 
    (W(i)).unsqueeze(0).mm((W(k)).unsqueeze(0).t()))
    opt = SGD(W.parameters(),lr=0.07,weight_decay=0.01)

    opt.zero_grad()
    print((W(i)).unsqueeze(0).mm((W(j)).unsqueeze(0).t()) , (W(i)).unsqueeze(0).mm((W(k)).unsqueeze(0).t()))
    print(y)
    y.backward()
    # print('grad',x,x.grad)
    opt.step()
    # print(F.logsigmoid(x))

thanks a lot~!

**Comment by SsnL** [comment]
> Theoretically,the loss will be lower in any condition , isn't it?

No. There is no theoretical guarantee unless you have some Lipschitz and smoothness guarantees.

**Comment by Cherrybruin** [comment]
> > Theoretically,the loss will be lower in any condition , isn't it?
> 
> No. There is no theoretical guarantee unless you have some Lipschitz and smoothness guarantees.

however, in the simple demo $loss=-logsigmoid(W(i)*W(j)-W(i)*W(k))$

assume 
$$
    W(X) = AX+B
$$
then,
$$
loss = -logsigmoid((Ai+B)^{T}·(A j+B)-(Ai+B)^{T}
·(Ak+B) )
$$

$$
 loss = -logsigmoid(  i^{T}A^{T}A(j-k)+B^{T}A(j-k)+i^{T}A^{T}B )
$$
grad,
$$
\frac{\partial loss}{\partial A} = (-logsigmoid)^{\prime}(  i^{T}A^{T}A(j-k)+B^{T}A(j-k)+i^{T}A^{T}B ) \odot \frac{\partial  i^{T}A^{T}A(j-k)+B^{T}A(j-k)+i^{T}A^{T}B}{\partial A}
$$
use Jacobian matrix or hessian matrix to get the grad?

**Comment by zou3519** [comment]
Please feel free to file us a new issue if this still persists.
